priam/src/main/java/com/netflix/priam/identity/token/TokenRetrieverUtils.java

```diff
@@ -74,14 +69,20 @@ public static String inferTokenOwnerFromGossip(
                 String ip = getIp(instance.getHostName(), token);
                 reachableInstances++;
 
-                if (StringUtils.isEmpty(ip)) {
-                    continue;
-                }
-
                 if (replaceIp == null) {
                     replaceIp = ip;
-                } else if (!replaceIp.equals(ip)) {
-                    break;
+                }
+
+                if (StringUtils.isEmpty(ip) || !replaceIp.equals(ip)) {
+                    // If the IP address produced by getIp call is empty it means it was able to
+                    // parse the status information and that token was still alive!!
+                    // We do not want to do anything if token is considered as alive by Cassandra.
+                    logger.info(
+                            "Not producing anything in replaceIp as according to C* that token is still alive or "
+                                    + "there is a mismatch in status information per Cassandra. ip: [{}], replaceIp: [{}]",
+                            ip,
+                            replaceIp);
+                    return null;
                 }
 
                 matchedGossipInstances++;
```

Comment:
`matchedGossipInstances` may not be needed anymore right? In case of a mismatch, we are returning null anyway.

Reply:
We need that so that we come out of the loop fast. As soon as matchedGossipInstances == desired (3 in most cases) we want to exit the loop.